certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -193,6 +198,8 @@ def choose_vhost(self, target_name):
         :rtype: :class:`~certbot_nginx.obj.VirtualHost`
 
         """
+        if self.vhost is not None:
```

Comment:
I think that's fine. We've done a few similar things in Apache. We can probably just use a dictionary that maps domain names to their SSLish server block(s).